From 93f8959715aac18ce2769bdd3e80434fc2258066 Mon Sep 17 00:00:00 2001
From: Yury Selivanov <yury@magic.io>
Date: Thu, 25 Apr 2019 11:53:45 -0400
Subject: [PATCH] Fix read_len_prefixed_{bytes|str}() to support 0-length
 sequences

---
 edb/server/pgproto/buffer.pyx | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/buffer.pyx b/buffer.pyx
index 1aedc72..aaedc0e 100644
--- a/edb/server/pgproto/buffer.pyx
+++ b/edb/server/pgproto/buffer.pyx
@@ -396,6 +396,8 @@ cdef class ReadBuffer:
         if size < 0:
             raise exceptions.BufferError(
                 'negative length for a len-prefixed bytes value')
+        if size == 0:
+            return b''
         return self.read_bytes(size)
 
     cdef str read_len_prefixed_utf8(self):
@@ -408,6 +410,9 @@ cdef class ReadBuffer:
             raise exceptions.BufferError(
                 'negative length for a len-prefixed bytes value')
 
+        if size == 0:
+            return ''
+
         self._ensure_first_buf()
         cbuf = self._try_read_bytes(size)
         if cbuf != NULL:
-- 
2.21.0

