From 743774f3da65c2d3035f32781c2bbb04f1918e79 Mon Sep 17 00:00:00 2001
From: Fantix King <fantix.king@gmail.com>
Date: Fri, 30 Aug 2024 13:34:32 -0400
Subject: [PATCH 6/6] Gate long STATEMENT log in SERIALIZATION error

---
 src/backend/utils/error/elog.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/backend/utils/error/elog.c b/src/backend/utils/error/elog.c
index 01066832e18..24bd20ce382 100644
--- a/src/backend/utils/error/elog.c
+++ b/src/backend/utils/error/elog.c
@@ -2384,6 +2384,11 @@ check_log_of_query(ErrorData *edata)
 	if (debug_query_string == NULL)
 		return false;
 
+	// Hide STATEMENT in serialization error if longer than 128 bytes
+	if (edata->sqlerrcode == ERRCODE_T_R_SERIALIZATION_FAILURE
+		&& strnlen(debug_query_string, 128) >= 128)
+		return false;
+
 	return true;
 }
 
-- 
2.46.0

