From 6999af9dc815624eebda741763972b0f2c88b458 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Tue, 16 Aug 2022 15:43:55 -0700
Subject: [PATCH] debug

---
 src/backend/utils/fmgr/dfmgr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backend/utils/fmgr/dfmgr.c b/src/backend/utils/fmgr/dfmgr.c
index e8c6cdde97..a028c0571f 100644
--- a/src/backend/utils/fmgr/dfmgr.c
+++ b/src/backend/utils/fmgr/dfmgr.c
@@ -244,8 +244,8 @@ internal_load_library(const char *libname)
 			/* errcode_for_file_access might not be appropriate here? */
 			ereport(ERROR,
 					(errcode_for_file_access(),
-					 errmsg("could not load library \"%s\": %s",
-							libname, load_error)));
+					 errmsg("could not load library \"%s\": %s; LD_LIBRARY_PATH=%s",
+							libname, load_error, getenv("LD_LIBRARY_PATH"))));
 		}
 
 		/* Check the magic function to determine compatibility */
-- 
2.35.1

