From a1f48b8e5f6595488c88f49e2b06d6ebe66ff90f Mon Sep 17 00:00:00 2001
From: Fantix King <fantix.king@gmail.com>
Date: Fri, 30 Aug 2024 13:34:32 -0400
Subject: [PATCH 6/6] Gate long STATEMENT log

---
 src/backend/utils/error/elog.c | 5 +++++
 1 file changed, 4 insertions(+)

diff --git a/src/backend/utils/error/elog.c b/src/backend/utils/error/elog.c
index 893f8a5b563..8bd84808a7d 100644
--- a/src/backend/utils/error/elog.c
+++ b/src/backend/utils/error/elog.c
@@ -2701,6 +2701,10 @@ check_log_of_query(ErrorData *edata)
 	if (debug_query_string == NULL)
 		return false;

+	// Hide STATEMENT if longer than 1024 bytes
+	if (strnlen(debug_query_string, 1024) >= 1024)
+		return false;
+
 	return true;
 }

--
2.46.0

