From 3f974817b98a3619189f03950cd359abfe9fc798 Mon Sep 17 00:00:00 2001
From: Fantix King <fantix.king@gmail.com>
Date: Thu, 14 Nov 2024 15:31:47 -0500
Subject: [PATCH 6/6] Gate long STATEMENT log

---
 src/backend/utils/error/elog.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/backend/utils/error/elog.c b/src/backend/utils/error/elog.c
index b924b524d0b..8068cde1198 100644
--- a/src/backend/utils/error/elog.c
+++ b/src/backend/utils/error/elog.c
@@ -2739,6 +2739,10 @@ check_log_of_query(ErrorData *edata)
 	if (debug_query_string == NULL)
 		return false;
 
+	// Hide STATEMENT if longer than 1024 bytes
+	if (strnlen(debug_query_string, 1024) >= 1024)
+		return false;
+
 	return true;
 }
 
-- 
2.47.0

