From 402c5a1b804760df5948dbbedfb6adac3aaf87cd Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Wed, 21 Apr 2021 14:59:44 -0700
Subject: [PATCH 5/5] Add ability to do pure arm64 builds

---
 configure    | 5 +++++
 configure.ac | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/configure b/configure
index 76fec6adbf..0b9938df41 100755
--- a/configure
+++ b/configure
@@ -8849,6 +8849,11 @@ $as_echo "$CC" >&6; }
                LIPO_INTEL64_FLAGS="-extract x86_64"
                ARCH_RUN_32BIT="true"
                ;;
+            arm-64)
+               UNIVERSAL_ARCH_FLAGS="-arch arm64"
+               LIPO_32BIT_FLAGS=""
+               ARCH_RUN_32BIT="true"
+               ;;
             intel)
                UNIVERSAL_ARCH_FLAGS="-arch i386 -arch x86_64"
                LIPO_32BIT_FLAGS="-extract i386"
diff --git a/configure.ac b/configure.ac
index 0b369958d1..2053f2ad52 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2380,6 +2380,11 @@ yes)
                LIPO_INTEL64_FLAGS="-extract x86_64"
                ARCH_RUN_32BIT="true"
                ;;
+            arm-64)
+               UNIVERSAL_ARCH_FLAGS="-arch arm64"
+               LIPO_32BIT_FLAGS=""
+               ARCH_RUN_32BIT="true"
+               ;;
             intel)
                UNIVERSAL_ARCH_FLAGS="-arch i386 -arch x86_64"
                LIPO_32BIT_FLAGS="-extract i386"
-- 
2.39.3

