From f76b59a8062c88b5fe09818837d4e1500b072afc Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Tue, 10 Aug 2021 19:51:36 -0700
Subject: [PATCH] Stop eagerly checking for OpenSSL rpath directory existence

---
 configure    | 7 +------
 configure.ac | 6 +-----
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/configure b/configure
index e79425d5b9..9849a94002 100755
--- a/configure
+++ b/configure
@@ -17766,12 +17766,7 @@ case $with_openssl_rpath in #(
   no) :
     OPENSSL_RPATH= ;; #(
   *) :
-    if test -d "$with_openssl_rpath"; then :
-  OPENSSL_RPATH="$with_openssl_rpath"
-else
-  as_fn_error $? "--with-openssl-rpath \"$with_openssl_rpath\" is not a directory" "$LINENO" 5
-fi
-
+    OPENSSL_RPATH="$with_openssl_rpath"
  ;;
 esac
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $OPENSSL_RPATH" >&5
diff --git a/configure.ac b/configure.ac
index 4c2ac5915a..5ce7595e71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5822,11 +5822,7 @@ AC_ARG_WITH(openssl-rpath,
 AS_CASE($with_openssl_rpath,
     [auto|yes],[OPENSSL_RPATH=auto],
     [no],[OPENSSL_RPATH=],
-    [AS_IF(
-        [test -d "$with_openssl_rpath"],
-        [OPENSSL_RPATH="$with_openssl_rpath"],
-        AC_MSG_ERROR([--with-openssl-rpath "$with_openssl_rpath" is not a directory]))
-    ]
+    [OPENSSL_RPATH="$with_openssl_rpath"]
 )
 AC_MSG_RESULT($OPENSSL_RPATH)
 AC_SUBST([OPENSSL_RPATH])
-- 
2.32.0

