diff --git a/dput/methods/sftp.py.orig b/dput/methods/sftp.py
index bd4ca6f..8e67750 100644
--- a/dput/methods/sftp.py.orig
+++ b/dput/methods/sftp.py
@@ -47,6 +47,13 @@ class ProcessAsChannelAdapter(object):
                 return ''
             raise
 
+    def recv_ready(self):
+        FIONREAD = 0x541B
+        import fcntl, struct, array
+        buf = array.array('l', [0])
+        fcntl.ioctl(self.__socket, FIONREAD, buf, 1)
+        return buf[0] > 0
+
     def close(self):
         """ Close and wait for process to finish. """
         self.__socket.close()
@@ -86,6 +93,7 @@ def copy_file(sftp_client, local_path, remote_path, debug, progress):
 
         # TODO: Do atomic?
         with sftp_client.file(remote_path, "w") as remote_fileobj:
+            remote_fileobj.set_pipelined()
             while True:
                 data = fileobj.read(4096)
                 if not data:
